internal/http/services/owncloud/ocdav/dav.go

```diff
@@ -142,6 +144,10 @@ func (h *DavHandler) Handler(s *svc) http.Handler {
 				return
 			}
 			log.Debug().Interface("statInfo", statInfo).Msg("Stat info from public link token path")
+			err = deletePublicShareIfExpired(ctx, s, token)
+			if err != nil {
+				w.WriteHeader(http.StatusInternalServerError)
```

Comment:
only log here, don't write a different header.

now if an expiration is not possible, the question is whether we should even allow accessing that share.
depending on the reason for the failed deletion, it is likely that the reason is either "shares data structure corrupt" or "service not available", so maybe it's ok to return an error, but then also call `return` ?

Reply:
delete this way should be a noop. I'll update it. I struggled with error handling with this one.